backend/clubs/views.py

```diff
@@ -513,6 +532,58 @@ def subscription(self, request, *args, **kwargs):
         serializer = SubscribeSerializer(Subscribe.objects.filter(club=club), many=True)
         return Response(serializer.data)
 
+    @action(detail=True, methods=["get"])
+    def analytics(self, request, *args, **kwargs):
+        """
+        Returns a list of all analytics (club visits, favorites,
+        subscriptions) for a club.
+        """
+        club = self.get_object()
+        if "date" in request.query_params.keys():
+            date = datetime.datetime.strptime(request.query_params["date"], "%Y-%m-%d")
+        else:
+            date = datetime.date.today()
+
+        visits_data = []
+        favorites_data = []
+        subscriptions_data = []
+        max_value = 0
+
+        for hour in range(24):
+            visits_data.append({"x": hour_to_string_helper(hour), "y": 0})
+            favorites_data.append({"x": hour_to_string_helper(hour), "y": 0})
+            subscriptions_data.append({"x": hour_to_string_helper(hour), "y": 0})
+
+        visits = ClubVisit.objects.filter(club=club, created_at__day=date.day)
```

Comment:
Ok I'm gonna leave this until later I think because my main priority is just having this be functional for SAC fair. This is something I will optimize in the future though because I have some ideas for it.